QEfficient/generation/text_generation_inference.py

```diff
@@ -167,6 +168,16 @@ def fix_prompts(prompt: List[str], batch_size: int):
                 "Number of prompts are not multiple of batch size, dropping last incomplete batch from given input prompts"
             )
             prompt = prompt[: batch_size * (len(prompt) // batch_size)]
+
+        if full_batch_size:
```

Comment:
This is not the right place for checking `full_batch_size` value, we should check it outside of this function.
And  can we have a separate function for `fixing_prompts_for_cb`  -> The logic for this function is same when `len(prompt)< full_batch_size`, whereas in else case, we don't need to truncate the prompts as cb can support it.

Reply:
we don't need to drop the prompts, in case of full_batch_size can we change the implementation to ->
```
def fix_prompts(prompt: List[str], batch_size: int, full_batch_size: int = None):
    if len(prompt) < batch_size or (full_batch_size is not None and len(prompt) < full_batch_size):
        logger.warning("Number of prompts are less than batch size/full batch size, repeating to required batch size")
        prompt = prompt * -(batch_size // -len(prompt))  # Repeat prompt to required size
        prompt = prompt[:batch_size]  # Truncate prompts to required size
    elif not full_batch_size:
        if (len(prompt) % batch_size) > 0:
            logger.warning(
                "Number of prompts are not multiple of batch size, dropping last incomplete batch from given input prompts"
            )
            prompt = prompt[: batch_size * (len(prompt) // batch_size)]

    return prompt
```
and test the condition of `batch_size=1` when `fbs` is not None outside this function? inside `cloud_ai_100_exec_kv`?